### PR TITLE
Fix/useAuth hook

### PR DESCRIPTION
**Pull request by mariaporcina**
handling auth error on login method

### CODE REVIEW

**Comment by mariaporcina** [comment]
I believe the execution stopped when an error occurs, so I used a catch to handle it.

here's an example of the usage of the "error" state:
image

File: examples/with-expo/hooks/useAuth.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ import {
 export default function useAuth() {
```
Comment (andresjesse) [inline examples/with-expo/hooks/useAuth.ts]: this is the wrong file (it's from the example, but the official hooks are in the root folder). I'll update it.

File: examples/with-expo/hooks/useAuth.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ import {
...
   const [loading, setLoading] = useState(true);
   const [user, setUser] = useState<User | null>(null);
+  const [error, setError] = useState(false);
```
Comment (andresjesse) [inline examples/with-expo/hooks/useAuth.ts]: Interesting approach! I've already seen something like that on UI libraries. The point is that (mainly) this hooks are throwing the errors when they happen, so I'll change just to keep this file using the same standard as the other ones.

File: examples/with-expo/hooks/useAuth.ts [hunk 3]
```diff
@@ -23,7 +24,12 @@ export default function useAuth() {
    */
   const login = async (email: string, password: string) => {
     setLoading(true);
-    await signInWithEmailAndPassword(getAuth(), email, password);
+    await signInWithEmailAndPassword(getAuth(), email, password)
```
Comment (andresjesse) [inline examples/with-expo/hooks/useAuth.ts]: What about this:

image

Comment (andresjesse) [reply examples/with-expo/hooks/useAuth.ts]: This way the error still being throwed and we have a guarantee to complete the loading

**Review by andresjesse — CHANGES_REQUESTED** [review]

**Review by andresjesse — APPROVED** [review]
I've refactored the PR.

File: examples/with-expo/hooks/useAuth.ts (hunk 1 above)
Comment (mariaporcina) [reply examples/with-expo/hooks/useAuth.ts]: oops, my bad hehe

File: examples/with-expo/hooks/useAuth.ts (hunk 3 above)
Comment (mariaporcina) [reply examples/with-expo/hooks/useAuth.ts]: nice!

**Pull request merged**